src/WebJobs.Extensions.DurableTask/MessagePayloadDataConverter.cs

```diff
@@ -13,31 +13,68 @@ namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
 {
     internal class MessagePayloadDataConverter : JsonDataConverter
     {
-        // The default JsonDataConverter for DTFx includes type information in JSON objects. This causes issues
-        // because the type information generated from C# scripts cannot be understood by DTFx. For this reason, explicitly
-        // configure the JsonDataConverter to not include CLR type information. This is also safer from a security perspective.
-        internal static readonly JsonSerializerSettings MessageSettings = new JsonSerializerSettings
-        {
-            TypeNameHandling = TypeNameHandling.None,
-        };
-
         private static readonly JsonSerializerSettings ErrorSettings = new JsonSerializerSettings
         {
             ContractResolver = new ExceptionResolver(),
             TypeNameHandling = TypeNameHandling.Objects,
             ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
         };
 
-        // Default singleton instances
-        public static readonly MessagePayloadDataConverter Default = new MessagePayloadDataConverter(MessageSettings);
-        public static readonly MessagePayloadDataConverter ErrorConverter = new MessagePayloadDataConverter(ErrorSettings);
-        public static readonly JsonSerializer DefaultSerializer = JsonSerializer.Create(MessageSettings);
+        private MessagePayloadDataConverter messageConverter;
+        private MessagePayloadDataConverter errorConverter;
+        private JsonSerializer messageSerializer;
+
+        public MessagePayloadDataConverter(ISerializerSettingsFactory serializerSettingsFactory)

```

Comment:
The other constructor called into the base constructor and passed in the appropriate settings. Don't we need to do that here as well? Otherwise it looks like it will use some other settings that it generates itself.